components/dashboard/UpcomingInterviewDialog/UpcomingInterviewDialog.js

```diff
@@ -0,0 +1,250 @@
+                    </Typography>
+                  )
+                }
+                value={values.type || ''}
+              >
+                {INTERVIEW_TYPES.map(type => (
+                  <MenuItem value={type} key={type.value} className={classes.selectItem}>
+                    <Typography variant="body1">{type.label}</Typography>
+                    <Typography variant="caption">{type.description}</Typography>
+                  </MenuItem>
+                ))}
+              </Select>
+              {!!errors.type && <FormHelperText>{errors.type}</FormHelperText>}
+            </FormControl>
+            <FormControl>
+              <MuiPickersUtilsProvider utils={DateFnsUtils}>
+                <KeyboardDatePicker
+                  disablePast
+                  disableToolbar
+                  error={!!errors.date}
+                  format="MM/dd/yyyy"
+                  helperText={errors.date}
+                  id={`${formId}-date`}
+                  InputLabelProps={{ shrink: true }}
+                  KeyboardButtonProps={{ 'aria-label': 'Interview Date' }}
+                  label="Interview Date"
+                  margin="normal"
+                  onChange={d => handleChangeCustom('date', d)}
+                  placeholder="When is your interview?"
+                  value={values.date || null}
+                  variant="inline"
+                />
+              </MuiPickersUtilsProvider>
+            </FormControl>
+            <TextField
+              error={!!errors.organization}
+              fullWidth
+              helperText={errors.organization}
+              id={`${formId}-organization`}
+              InputLabelProps={{ shrink: true }}
+              inputProps={{ name: 'organization' }}
+              label="Organization"
+              margin="normal"
+              onChange={handleChange}
+              placeholder="Who is the interview with?"
+              value={values.organization}
+            />
+            <TextField
+              error={!!errors.role}
+              fullWidth
+              helperText={errors.role}
+              id={`${formId}-role`}
+              InputLabelProps={{ shrink: true }}
+              inputProps={{ name: 'role' }}
+              label="Role / Position"
+              margin="normal"
+              onChange={handleChange}
+              placeholder="What role did you apply for?"
+              value={values.role}
+            />
+          </form>
+        )}
+      </DialogContent>
+      <DialogActions>
+        {success ? (
+          <Button onClick={onClose} color="primary">
+            Close
+          </Button>
+        ) : (
+          <Button onClick={handleSubmit} color="primary">
+            Let Us Know
+          </Button>
+        )}
+        {submitError && (
+          <Typography color="error" variant="h6">
+            Error: {submitError}
+          </Typography>
+        )}
+      </DialogActions>
+      <Box
```

Comment:
This could probably be moved into a separate component down the line (loading overlay)

Reply:
👍 0, 1, ∞